.gitbook/assets/ReadonlyREST-api.yml

```diff
@@ -0,0 +1,209 @@
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UpdateMainSettingsRequest'
+      responses:
+        200:
+          description: Returns a status of updating Main Settings
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/UpdateMainSettingsResponse'
+              examples:
+                UpdateMainSettingsSuccessResponse:
+                  $ref: "#/components/examples/UpdateMainSettingsSuccessResponse"
+                UpdateMainSettingsFailedResponse:
+                  $ref: "#/components/examples/UpdateMainSettingsFailedResponse"
+        403:
+          description: The request was forbidden by ReadonlyREST's ACL
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ForbiddenRorResponse'
+              examples:
+                NoAclBlockWasMatchedForbiddenResponseExample:
+                  $ref: "#/components/examples/NoAclBlockWasMatchedForbiddenResponseExample"
+                ForbidBlockMatchedResponseExample:
+                  $ref: "#/components/examples/ForbidBlockMatchedResponseExample"
+                TestSettingsNotConfiguredForbiddenResponseExample:
+                  $ref: "#/components/examples/TestSettingsNotConfiguredForbiddenResponseExample"
+                ImpersonationNotSupportedResponseExample:
+                  $ref: "#/components/examples/ImpersonationNotSupportedResponseExample"
+                ImpersonationNotAllowedResponseExample:
+                  $ref: "#/components/examples/ImpersonationNotAllowedResponseExample"
+                RorIsNotStartedYetResponseExample:
+                  $ref: "#/components/examples/RorIsNotStartedYetResponseExample"
+                RorFailedToStartResponseExample:
+                  $ref: "#/components/examples/RorFailedToStartResponseExample"
+                CannotSaveClusterWideSettingsOnFreeLicense:
+                  $ref: "#/components/examples/CannotSaveClusterWideSettingsOnFreeLicense"
+
+components:
+  securitySchemes:
+    http:
+      type: http
+      scheme: basic
+
+  schemas:
+    ForbiddenRorResponse:
+      type: object
+      required:
+        - error
+        - status
+      properties:
+        error:
+          type: object
+          required:
+            - type
+            - reason
+            - due_to
+          properties:
+            type:
+              type: string
+              enum:
+                - forbidden_response
+            reason:
+              type: string
+              enum:
+                - forbidden
+            due_to:
+              type: array
+              description: |
+                It's an array of causes why ReadonlyREST's ACL forbade the request. 
+                WARNING: Please notice, that one element array ES used to print as a string!!!
```

Comment:
cannot we "fix" that?

Reply:
It was redundant, removed ✔️

Reply:
"used to print" like when? and now?

Reply:
ES JSON library (Jackson?) renders one element array like this:
```
{ 
  array_filed: 1
}
```

two elements array looks like this:
```
{
  array_filed: [1, 2]
}
```

I don't know why it's done like that. Maybe this is a bug or sth - for sure the problem lies in the ES internals. 

Nevertheless, IMO we should NOT imitate ES behaviour. Let's do it as JSON spec says (I mean, let's handle it internally on the ROR KBN side)